forge-gui/res/cardsfolder/upcoming/unexplained_absence.txt

```diff
@@ -0,0 +1,9 @@
+Name:Unexplained Absence
+ManaCost:3 W
+Types:Instant
+A:SP$ ChangeZone | Origin$ Battlefield | Destination$ Exile | ValidTgts$ Permanent.nonLand | TgtPrompt$ Select up to one target nonland permanent each player controls | TargetMin$ 0 | TargetMax$ OneEach | TargetsWithDifferentControllers$ True | SubAbility$ LoopTgts | StackDescription$ SpellDescription | SpellDescription$ For each player, exile up to one target nonland permanent that player controls. For each permanent exiled this way, its controller cloaks the top card of their library.
+SVar:LoopTgts:DB$ RepeatEach | RepeatSubAbility$ CloakEach | DefinedCards$ Targeted
```

Comment:
why the extra RepeatEach?

Reply:
This is borrowed from Luminate Primordial's script, just makes sure only the players who had a permanent exiled this way can cloak a card from library. Couldn't find a more modern example with this effect which is more concise

Reply:
- _Luminate Primordial_ probably needs it because the amount for each player is variable
- in the current version you wouldn't even need the cleanup due to internal handling by RepeatEach
- However you should use RememberLKI$ since it's explicitly stating "exiled this way"
- The CloackEach line will work fine by itself then

Reply:
Removed the RepeatEach and added RememberLKI to the script, works fine! Thank you